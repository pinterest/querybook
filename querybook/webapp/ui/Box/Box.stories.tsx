import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import centered from '@storybook/addon-centered/react';

import { Box } from './Box';

export default {
    title: 'Layout/Box',
    decorators: [centered],
};

export const _Box = ({ ...args }) => <Box {...args} />;
_Box.args = {
    children: 'Box',
};
_Box.argTypes = {
    children: { control: 'text' },
};
