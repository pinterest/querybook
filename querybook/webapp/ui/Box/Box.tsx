import styled from 'styled-components';

export const Box = styled.div.attrs({
    className: 'Box',
})`
    background-color: var(--bg-color);
    box-shadow: var(--box-shadow);
    padding: 18px;
    border-radius: var(--border-radius);
`;
