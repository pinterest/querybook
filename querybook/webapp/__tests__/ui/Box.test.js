import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';

import { Box } from '../../ui/Box/Box';

it('renders without crashing', () => {
    shallow(<Box>Test</Box>);
});

describe('matches enzyme snapshots', () => {
    it('matches snapshot', () => {
        const wrapper = shallow(<Box>Test</Box>);
        const serialized = toJson(wrapper);
        expect(serialized).toMatchSnapshot();
    });
});

describe('matches test renderer snapshot', () => {
    it('serializes the styles', () => {
        const output = renderer.create(<Box>Test</Box>);
        expect(output).toMatchSnapshot();
    });
});
