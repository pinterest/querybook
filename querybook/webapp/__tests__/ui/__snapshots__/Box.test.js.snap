// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches enzyme snapshots matches snapshot 1`] = `
<div
  className="Box-sc-11jncwz-0 UciDk Box"
>
  Test
</div>
`;

exports[`matches test renderer snapshot serializes the styles 1`] = `
<div
  className="Box-sc-11jncwz-0 UciDk Box"
>
  Test
</div>
`;
